apps/studio/components/interfaces/Settings/Database/DatabaseSettings/DatabaseConnectionString.tsx

```diff
@@ -218,7 +225,13 @@ export const DatabaseConnectionString = ({ appearance }: DatabaseConnectionStrin
                 onCheckedChange={snap.setUsePoolerConnection}
                 onSelectPoolingMode={setPoolingMode}
               />
-              {!snap.usePoolerConnection && <IPv4DeprecationNotice />}
+              {isSuccessAddons &&
+                poolingMode === 'session' &&
+                ipv4Addon !== undefined &&
+                snap.usePoolerConnection && <IPv4AddonDirectConnectionNotice />}
+              {isSuccessAddons
+                ? ipv4Addon === undefined && !snap.usePoolerConnection && <IPv4DeprecationNotice />
+                : !snap.usePoolerConnection && <IPv4DeprecationNotice />}
```

Comment:
this looks like it can be just like this? since in both ways we're checking `!snap.usePoolerConnection`?

```suggestion
              {isSuccessAddons && !snap.usePoolerConnection && <IPv4DeprecationNotice />}
```

Reply:
so here i was trying to make sure that the user would see some notice in case the call to add ons failed.

If call to add ons succeed then i want to check `ipv4Addon === undefined` as part of it.

can simplify it to `ipv4Addon === undefined && !snap.usePoolerConnection && <IPv4DeprecationNotice />`, similar to what you suggested, if we don't care about `isSuccessAddons` when it's `false`.

Reply:
that's how i had it here: https://github.com/supabase/supabase/pull/26903/files#diff-0affa75a26bfa91b825973f555005deb19360c715009b7647ffb9370ac70ff7cR144

Reply:
ahh okay i see what you mean here

Reply:
^ yeah i think we don't need to check for isSuccessAddons here, just `ipv4Addon === undefined && !snap.usePoolerConnection && <IPv4DeprecationNotice />` as you mentioned will work - i'll make this change

Reply:
Sounds good thanks